app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala

```diff
@@ -23,23 +23,27 @@ import play.api.mvc.Results.Forbidden
 import play.api.mvc.{ActionRefiner, Result}
 import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.{ParsedXmlRequest, ValidatedXmlRequest}
 import uk.gov.hmrc.excisemovementcontrolsystemapi.models.{EmcsUtils, ErrorResponse}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.MessageService
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class ValidateErnsActionImpl @Inject()(implicit val executionContext: ExecutionContext, implicit val emcsUtils: EmcsUtils)
-  extends ValidateErnsAction
-    with Logging {
+class ValidateErnsActionImpl @Inject()(messageService: MessageService)(
+  implicit val executionContext: ExecutionContext,
+  implicit val emcsUtils: EmcsUtils
+) extends ValidateErnsAction
+  with Logging {
   override def refine[A](request: ParsedXmlRequest[A]): Future[Either[Result, ValidatedXmlRequest[A]]] = {
 
-    val matchedErns: Set[String] = request.erns.intersect(request.ieMessage.getErns)
+    val matchedErns: Future[Set[String]] =
+      messageService.getErns(request.ieMessage).map(ernsFromMessage => request.erns.intersect(ernsFromMessage))
```

Comment:
is there any reason that this return a feature? I do not see any reason that this is a feature as we only want to take the list of valid ern from an IEMessage

Reply:
Yeah so for the IE810 we have to get the ERNs from the database so it returns a Future so the GetErns then returns a Future and then the intersection of the two sets has to be a Future - its futures all the way up

Reply:
already removed that in the last commit. (not sure why it took 2 hours for github to tell me you responded!)